### PR TITLE
Removed assets_by_site from the datastore

### PR DESCRIPTION
**Pull request by micheles**
Closes https://github.com/gem/oq-risklib/issues/368

### CODE REVIEW

File: openquake/calculators/scenario_risk.py [hunk 1]
```diff
@@ -82,20 +82,6 @@ def scenario_risk(riskinputs, riskmodel, rlzs_assoc, monitor):
     return result
 
 
-def check_time_event(dstore):
-    """
```
Comment (ptormene) [inline openquake/calculators/scenario_risk.py]: I see this has been moved, so I guess we already have a test for it. If we haven't, maybe we should add it.

File: openquake/calculators/base.py [hunk 2]
```diff
@@ -410,16 +432,30 @@ def read_risk_data(self):
             self.load_riskmodel()
             self.sitecol = haz_sitecol
 
+        if oq_hazard:
+            # TODO: move check_time_event outside the if
```
Comment (ptormene) [inline openquake/calculators/base.py]: The TODO is a bit cryptic

File: openquake/risklib/riskinput.py [hunk 3]
```diff
@@ -92,7 +141,7 @@ def build_asset_collection(assets_by_site, time_event=None):
                         name, lt = field.split('~')
                     except ValueError:  # no ~ in field
                         name, lt = 'value', field
-                    value = getattr(asset, name)(lt)
+                    value = getattr(asset, name + 's')[lt]
```
Comment (ptormene) [inline openquake/risklib/riskinput.py]: Please add a comment

File: openquake/calculators/base.py (hunk 2 above)
Comment (micheles) [inline openquake/calculators/base.py]: Removed

File: openquake/risklib/riskinput.py (hunk 3 above)
Comment (micheles) [inline openquake/risklib/riskinput.py]: Added

File: openquake/calculators/scenario_risk.py (hunk 1 above)
Comment (micheles) [inline openquake/calculators/scenario_risk.py]: The test is scenario_risk_test, see below

**Comment by ptormene** [comment]
LGTM

**Pull request merged**